onnx/defs/nn/defs.cc

```diff
@@ -3848,4 +3848,302 @@ ONNX_OPERATOR_SET_SCHEMA(
+```
+xSkip = (scaling_factor * input) + F(input) + Bias
+```
+where,
+F(input): denotes the output of a particular layer.
+scaling_factor: a modulating scalar that adjusts the importance of the skip.
+Bias: a bias term added to the output of the skip connection.
+
+LayerNorm is then applied to xSkip as follows:
+```
+output = LayerNormalization(xSkip)
+```
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    SkipLayerNormalization,
+    24,
+    OpSchema()
+        .SetDoc(SkipLayerNormalization_ver24_doc)
+        .Attr(
+            "axis",
+            "The dimension for layer normalization. If rank(X) is r, axis' allowed range is [-r, r). "
+            "Negative value means counting dimensions from the back.",
+            AttributeProto::INT,
+            static_cast<int64_t>(-1))
+        .Attr("epsilon", "The epsilon value to use to avoid division by zero.", AttributeProto::FLOAT, 1e-5f)
+        .Attr(
+            "scaling_factor",
+            "Modulating scalar by which the skip input is multiplied.",
+            AttributeProto::INT,
+            static_cast<int64_t>(1))
+        .Input(
+            0,
+            "X",
+            "The output of the layer for which the skip connection is being created. "
+            "In general, the shape is (N, C, D1, D2, ... , Dn) for n-dimensional data, where "
+            "D1 to Dn are the spatial dimension sizes and N is the batch size, C is the number of channels.",
+            "T")
+        .Input(
+            1,
+            "S",
+            "Skip input with same shape as X. This is the input to the layer for which the skip connection is being created.",
+            "T")
+        .Input(
+            2,
+            "gamma",
+            "1D tensor representing scale input of layer normalization "
+            "with shape of the spatial dimension along which layer normalization is applied.",
+            "T")
+        .Input(
+            3,
+            "beta",
+            "1D tensor representing bias input of layer normalization "
+            "with shape of the spatial dimension along which layer normalization is applied.",
+            "T",
+            OpSchema::Optional)
+        .Input(
+            4,
+            "B",
+            "1D bias tensor for the skip connection with shape of the spatial dimension along which layer normalization is applied.",
+            "T",
+            OpSchema::Optional)
+        .Output(0, "Y", "Output tensor with same shape as X", "T")
+        .Output(
+            1,
+            "InputSkipBiasSum",
+            "Sum of the input and skip inputs (and bias if it exists). Same shape as X",
+            "T",
+            OpSchema::Optional)
+        .TypeConstraint(
+            "T",
+            {"tensor(float)", "tensor(float16)"},
+            "Constrain input and output types to float or half tensors.")
+        .TypeConstraint("U", {"tensor(float)"}, "Constrain mean and inv_std_var to float tensors.")
```

Comment:
I feel like this should be aligned with the type constraints of the normal LayerNormalization. 
https://onnx.ai/onnx/operators/onnx__LayerNormalization.html#type-constraints
`bfloat16` in particular, and just for completeness `double` although I don't see the need for double.

Reply:
Following up on this point, we are also missing the `stash_type` attribute of LN. My concern is this ends up essentially duplicating a lot of the apparatus of our existing normalization ops, and any future changes to the normalization ops would need to be reflected back in these skip norm ops too.

Do you know what's the main use case of these ops and what was the motivation they were introduced as contrib ops? If the motivation is to suggest useful fusion patterns, would it be simple enough for backends to handle this with pattern matching?